src/integrations/watchers/indexable-home-url-option-watcher.php

```diff
@@ -0,0 +1,133 @@
+	 * @param Indexable_Helper $indexable The indexable helper.
+	 */
+	public function __construct( Post_Type_Helper $post_type, Options_Helper $options, Indexable_Helper $indexable ) {
+		$this->post_type        = $post_type;
+		$this->options_helper   = $options;
+		$this->indexable_helper = $indexable;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function register_hooks() {
+		\add_action( 'update_option_home', [ $this, 'reset_permalinks' ] );
+	}
+
+	/**
+	 * Resets the permalinks for everything that is related to the permalink structure.
+	 */
+	public function reset_permalinks() {
+		$post_types = $this->get_post_types();
+		foreach ( $post_types as $post_type ) {
+			$this->reset_permalinks_post_type( $post_type );
+		}
+
+		$taxonomies = $this->get_taxonomies_for_post_types( $post_types );
+		foreach ( $taxonomies as $taxonomy ) {
+			$this->indexable_helper->reset_permalink_indexables( 'term', $taxonomy, Indexation_Permalink_Warning_Presenter::REASON_HOME_URL_OPTION );
+		}
+
+		$this->indexable_helper->reset_permalink_indexables( 'home-page', null, Indexation_Permalink_Warning_Presenter::REASON_HOME_URL_OPTION );
+		$this->indexable_helper->reset_permalink_indexables( 'user', null, Indexation_Permalink_Warning_Presenter::REASON_HOME_URL_OPTION );
+		$this->indexable_helper->reset_permalink_indexables( 'date-archive', null, Indexation_Permalink_Warning_Presenter::REASON_HOME_URL_OPTION );
+		$this->indexable_helper->reset_permalink_indexables( 'system-page', null, Indexation_Permalink_Warning_Presenter::REASON_HOME_URL_OPTION );
+	}
+
+	/**
+	 * Resets the permalink for the given post type.
+	 *
+	 * @param string $post_type The post type to reset.
+	 */
+	public function reset_permalinks_post_type( $post_type ) {
+		$this->indexable_helper->reset_permalink_indexables( 'post', $post_type, Indexation_Permalink_Warning_Presenter::REASON_HOME_URL_OPTION );
+		$this->indexable_helper->reset_permalink_indexables( 'post-type-archive', $post_type, Indexation_Permalink_Warning_Presenter::REASON_HOME_URL_OPTION );
+	}
+
+	/**
+	 * Retrieves a list with the public post types.
+	 *
+	 * @return array The post types.
+	 */
+	protected function get_post_types() {
+		/**
+		 * Filter: Gives the possibility to filter out post types.
+		 *
+		 * @param array $post_types The post type names.
+		 *
+		 * @return array The post types.
+		 */
+		$post_types = \apply_filters( 'wpseo_post_types_reset_permalinks', $this->post_type->get_public_post_types() );
+
+		return $post_types;
+	}
+
+	/**
+	 * Retrieves the taxonomies that belongs to the public post types.
+	 *
+	 * @param array $post_types The post types to get taxonomies for.
+	 *
+	 * @return array The retrieved taxonomies.
+	 */
+	protected function get_taxonomies_for_post_types( $post_types ) {
+		$taxonomies = [];
+		foreach ( $post_types as $post_type ) {
+			$taxonomies[] = \get_object_taxonomies( $post_type, 'names' );
+		}
+
+		$taxonomies = \array_merge( [], ...$taxonomies );
```

Comment:
It shouldn't be spread operator on array but argument unpacking which is in PHP since 5.6
actually the very same code is elsewhere in `master`:
https://github.com/Yoast/wordpress-seo/blob/7b0a6942d23d5f609f81ea4ca056fcbc86887ae0/src/integrations/watchers/indexable-permalink-watcher.php#L183

Reply:
We went back and forth on this comment and have reached the conclusion that this isn't a breaking approach.